### PR TITLE
feat: custom user agent InternalCognitoUserPool

### PR DESCRIPTION
**Pull request by erinleigh90**
#### Description of changes
- Creates and exports InternalCognitoUserPool class from /internals scope with added userAgentValue parameter on public API's that result in service calls
- Sends received userAgentValue through to cognito client service calls

NOTES
- To see the changes to InternalCognitoUser after moving the changes over from CognitoUser, see [this comparison](https://github.com/erinleigh90/amplify-js/compare/5fa456c758ee46a49cab92fc8f22c7609ef7d4d2...e45491d0df658e0da7565d22a78a5ed823512f65#diff-9195874b28a5d2b0bab5e3dcaa863e337f5b5e5e98655f594e75dad41045e872)

#### Description of how you validated changes
- [x] unit tests pass
- [x] integ tests pass
- [x] smoke tested in sample app

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x]  Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/src/Client.js [hunk 1]
```diff
@@ -75,11 +75,11 @@ export default class Client {
 	 * @param {function} callback Callback called when a response is returned
```
Comment (jimblanc) [inline packages/amazon-cognito-identity-js/src/Client.js:75]:
```suggestion
	 * @param {function} callback Callback called when a response is returned
	 * @param {string} userAgentValue Optional string containing custom user agent value
```

File: packages/amazon-cognito-identity-js/src/internals/InternalCognitoUserPool.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
+/*!
```
Comment (jimblanc) [inline packages/amazon-cognito-identity-js/src/internals/InternalCognitoUserPool.js:1]:
```suggestion
/*
```

**Review by jimblanc — APPROVED** [review]
LGTM, two nits below

**Review by jimblanc — DISMISSED** [review]

**Review by israx — DISMISSED** [review]
LGTM

File: packages/amazon-cognito-identity-js/__tests__/internalsIndex.test.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ describe('import * keys', () => {
 		Array [
 		  "addAuthCategoryToCognitoUserAgent",
 		  "addFrameworkToCognitoUserAgent",
+		  "InternalCognitoUserPool",
```
Comment (stocaaro) [inline packages/amazon-cognito-identity-js/__tests__/internalsIndex.test.js:9]: nit: I know the only diff here is arity and exposing the extended interface for internals. Is it worth adding test coverage for the internals surface?

**Review by stocaaro — DISMISSED** [review]

File: packages/amazon-cognito-identity-js/__tests__/internalsIndex.test.js (hunk 3 above)
Comment (erinleigh90) [reply packages/amazon-cognito-identity-js/__tests__/internalsIndex.test.js:9]: When I originally added this test as a part of the cognitoUserAgent changes, I was attempting to reflect the other test class on main auth exports. However, it does seem rather redundant and we probably don't need it. I will remove the test.

**Review by israx — APPROVED** [review]

**Pull request merged**